### DISCUSSION TITLE
should core/parser be exported?

### DISCUSSION THREAD

**Discussion by jameyg42** [oclif/core#1097] [General]
Somewhat under the "Everything is Customizable" feature, prior to v4 the core/parser module could be used/customized standalone.  Since v4 this is no longer possible since `core/parser` (and `core/interfaces/parser.js`) is not exported.  

One example use case is the use of `Parser.parse()` without calling `validate()` to allow for "unrecognized flags" similar to how `strict=false` allows for unrecognized args.

A specific example is if an `oclif` command is simply wrapping another command and we want to pass the wrapped command its arguments w/out having to understand them.  This could previously be accomplished by using `Parser.parse()` as-is, treating any detected "nonexistent flag" simply as "unrecognized", then calling `validate()` w/ an empty nonexistent-flags array.  While `--` would work, it ends up defeating the illusion that we're "extending" another (non-oclif) command.

could/should `core/parser` be exported?

**Comment by mdonnalley** [comment]
Sorry for the oversight. I can make a PR to add parser to the exports

https://github.com/oclif/core/pull/1098

**Comment by jameyg42** [comment]
It might be more than this (#1098).  For example, right now we can't easily override something like core/Command.parse() because all of the types are in the unexported core/interfaces/parser.  But i realize what I'm probably asking for (which likely makes it too large of an ask)  would probably be closer to exporting core/interfaces/*.js and core/parser/*.js

**Reply by mdonnalley** [reply]
Sorry, I didn't read through your original post more closely - too much multitasking today! I'll reopen the discussion

I think we stopped exporting those types because it made it difficult to improve the type safety of flags without introducing breaking changes for anyone who might be using importing types. I'm open to exporting them again though since that part of the code base has stabilized over the past year or so

**Comment by mdonnalley** [comment]
I exported all the relevant types in https://github.com/oclif/core/pull/1099

Feel free to make a new issue or PR to add anything else you need
